swift/ql/test/query-tests/Security/CWE-022/testPathInjection.swift

```diff
@@ -100,65 +142,75 @@ func test() {
+    let _ = fm.fileExists(atPath: remoteString, isDirectory: UnsafeMutablePointer<ObjCBool>.init(bitPattern: 0)) // $ hasPathInjection=139
+    fm.setAttributes([:], ofItemAtPath: remoteString) // $ hasPathInjection=139
+    let _ = fm.contents(atPath: remoteString) // $ hasPathInjection=139
+    let _ = fm.contentsEqual(atPath: remoteString, andPath: "") // $ hasPathInjection=139
+    let _ = fm.contentsEqual(atPath: "", andPath: remoteString) // $ hasPathInjection=139
+    let _ = fm.changeCurrentDirectoryPath(remoteString) // $ hasPathInjection=139
+    let _ = fm.unmountVolume(at: remoteUrl, options: [], completionHandler: { _ in }) // $ hasPathInjection=139
     // Deprecated methods
-    let _ = fm.changeFileAttributes([:], atPath: remoteString) // $ hasPathInjection=97
-    let _ = fm.directoryContents(atPath: remoteString) // $ hasPathInjection=97
-    let _ = fm.createDirectory(atPath: remoteString, attributes: [:]) // $ hasPathInjection=97
-    let _ = fm.createSymbolicLink(atPath: remoteString, pathContent: "") // $ hasPathInjection=97
-    let _ = fm.createSymbolicLink(atPath: "", pathContent: remoteString) // $ hasPathInjection=97
-    let _ = fm.pathContentOfSymbolicLink(atPath: remoteString) // $ hasPathInjection=97
-    let _ = fm.replaceItemAtURL(originalItemURL: remoteNsUrl, withItemAtURL: safeNsUrl, backupItemName: nil, options: []) // $ hasPathInjection=97
-    let _ = fm.replaceItemAtURL(originalItemURL: safeNsUrl, withItemAtURL: remoteNsUrl, backupItemName: nil, options: []) // $ hasPathInjection=97
+    let _ = fm.changeFileAttributes([:], atPath: remoteString) // $ hasPathInjection=139
+    let _ = fm.directoryContents(atPath: remoteString) // $ hasPathInjection=139
+    let _ = fm.createDirectory(atPath: remoteString, attributes: [:]) // $ hasPathInjection=139
+    let _ = fm.createSymbolicLink(atPath: remoteString, pathContent: "") // $ hasPathInjection=139
+    let _ = fm.createSymbolicLink(atPath: "", pathContent: remoteString) // $ hasPathInjection=139
+    let _ = fm.pathContentOfSymbolicLink(atPath: remoteString) // $ hasPathInjection=139
+    let _ = fm.replaceItemAtURL(originalItemURL: remoteNsUrl, withItemAtURL: safeNsUrl, backupItemName: nil, options: []) // $ hasPathInjection=139
+    let _ = fm.replaceItemAtURL(originalItemURL: safeNsUrl, withItemAtURL: remoteNsUrl, backupItemName: nil, options: []) // $ hasPathInjection=139
+
+    NSString().write(to: remoteUrl, atomically: true, encoding: 0)  // $ hasPathInjection=139
+    NSString().write(toFile: remoteString, atomically: true, encoding: 0) // $ hasPathInjection=139
+    let _ = NSKeyedUnarchiver().unarchiveObject(withFile: remoteString) // $ hasPathInjection=139
+    let _ = ArchiveByteStream.fileStream(fd: remoteString as! FileDescriptor, automaticClose: true) // $ hasPathInjection=139
+    ArchiveByteStream.withFileStream(fd: remoteString as! FileDescriptor, automaticClose: true) { _ in } // $ hasPathInjection=139
```

Comment:
Where is the implementation used by `remoteString as! FileDescriptor`?

Reply:
Do you mean the functions being called? They're stubbed [here](https://github.com/github/codeql/pull/11742/files#diff-0c75be2b1a02487e04ac197b1e372a3db6ac7b46507adcef91049698add3064fR125-R126).

Reply:
Yes, it's a forced cast to keep it simple. That would probably not work if the code was actually executed, but it should be good enough for the test.